.github/workflows/bcny-build-nuget-package.yml

```diff
@@ -0,0 +1,142 @@
+name: "bcny-build-nuget-package"
+on:
+  workflow_call:
+    inputs:
+      arch:
+        description: "Name of the architecture amd64 or arm64"
+        required: true
+        type: string
+      platform:
+        description: "MSVC understandable platform name x64 or ARM64"
+        required: true
+        type: string
+
+jobs:
+  build_and_package:
+    name: "Build and package libsodium"
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: compnerd/gha-setup-vsdevenv@main
+        with:
+          host_arch: amd64
+          components: 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64;Microsoft.VisualStudio.Component.VC.Tools.ARM64'
+          arch: ${{ inputs.arch }}
+
+      - uses: actions/setup-python@v4
+        id: python
+        with:
+          python-version: 3.9
+          architecture: 'x64'
+
+      - name: Build StaticRelease
+        run:
+          msbuild .\builds\msvc\vs2022\libsodium.sln /p:Configuration=StaticRelease /p:Platform=${{ inputs.platform }}
+
+      - name: Build DynRelease
+        run:
+          msbuild .\builds\msvc\vs2022\libsodium.sln /p:Configuration=DynRelease /p:Platform=${{ inputs.platform }}
+
+      - name: Build LtcgRelease
+        run:
+          msbuild .\builds\msvc\vs2022\libsodium.sln /p:Configuration=LtcgRelease /p:Platform=${{ inputs.platform }}
+
+      - name: Package libsodium
+        run: |
+          @"
+          <?xml version="1.0" encoding="utf-8"?>
+          <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+            <metadata>
+              <id>org.libsodium.libsodium.windows.${{ inputs.arch }}</id>
+              <version>0.0.0</version>
+              <title>libsodium </title>
+              <description>libsodium</description>
+              <authors>org.libsodium</authors>
+              <projectUrl>https://libsodium.org</projectUrl>
+              <repository type="git" url="https://github.com/jedisct1/libsodium" branch="master" />
+            </metadata>
+            <files>
+              <file src="${{ github.workspace }}\bin\${{inputs.platform}}\Release\v143\static\libsodium.lib" target="lib/static" />
+              <file src="${{ github.workspace }}\bin\${{inputs.platform}}\Release\v143\dynamic\libsodium.lib" target="lib/dynamic" />
+              <file src="${{ github.workspace }}\bin\${{inputs.platform}}\Release\v143\dynamic\libsodium.pdb" target="lib/dynamic" />
+              <file src="${{ github.workspace }}\bin\${{inputs.platform}}\Release\v143\dynamic\libsodium.dll" target="lib/dynamic" />
```

Comment:
Lets hold off on the shared version until we deem it necessary.

Reply:
Sure